src/js/components/FileInput/FileInput.js

```diff
@@ -357,7 +377,8 @@ const FileInput = forwardRef(
                   align="center"
                   direction="row"
                 >
-                  {maxSize && file.size > maxSize && <CircleAlert />}
+                  {(maxSize || max) &&
+                    (file.size > maxSize || index >= max) && <CircleAlert />}
```

Comment:
I suggest this for readability and to avoid a corner-case:
```jsx
{((maxSize && file.size > maxSize) ||
  (max && index >= max)) && <CircleAlert />}
```
The corner case is if `max` is passed in as an empty string and there's a `maxSize` then `index >= max` is actually true and you can get an erroneous CircleAlert icon on the items.